docs/docs/guides/migrate/sources/auth0-guide.md

```diff
@@ -0,0 +1,186 @@
+
+---
+
+## 4. Running the ZITADEL Migration Tool
+
+### 4.1. Install the Migration Tool
+Follow the installation instructions to set up the ZITADEL migration tool from [ZITADEL Tools](https://github.com/zitadel/zitadel-tools?tab=readme-ov-file#installation).
+
+### 4.2. Generate Import JSON
+Use the migration tool to convert the Auth0 export file to a ZITADEL-compatible JSON.
+Step-by-step instructions: [Migration Tool for Auth0](https://github.com/zitadel/zitadel-tools/blob/main/cmd/migration/auth0/readme.md)
+
+Typical steps:
+- Run the migration tool with your exported Auth0 file as input.
+- The tool generates a JSON file ready for import into ZITADEL.
+
+Example:
+After obtaining the 2 required input files (passwords and profile) in JSON lines format, you can run the following command:
+
+Sample `passwords.ndjson` content:
+```json
+{"_id":{"$oid":"emxdpVxozXeFb1HeEn5ThAK8"},"email_verified":true,"email":"tommie_krajcik85@hotmail.com","passwordHash":"$2b$10$d.GvZhGwTllA7OdAmsA75uGGzqr/mhdQoU88M3zD.fX3Vb8Rcf33.","password_set_date":{"$date":"2025-06-30T00:00:00.000Z"},"tenant":"test","connection":"Username-Password-Authentication","_tmp_is_unique":true}
+```
+
+Sample `profiles.ndjson` content:
+```json
+{"user_id":"auth0|emxdpVxozXeFb1HeEn5ThAK8","email_verified":true,"name":"Tommie Krajcik","email":"tommie_krajcik85@hotmail.com"}
+```
+
+Run the following command in your terminal:
+```bash
+zitadel-tools migrate auth0 --org=<ORG_ID> --users=./profiles.ndjson --passwords=./passwords.ndjson --multiline --email-verified --output=./importBody.json
+```
+
+The tool will merge both objects into a single one in the importBody.json output, this will be used in the next step to complete the import process.
+
+## 5. Importing Users into ZITADEL
+
+### 5.1. Obtain Access Token (or PAT) for API Access
+
+To call the ZITADEL Admin API, you need to authenticate using a **Service User** with the `IAM_OWNER` Manager permissions.
+
+There are two recommended authentication methods:
+
+- **Client Credentials Flow**  
+  [Learn how to authenticate with client credentials.](https://zitadel.com/docs/guides/integrate/service-users/client-credentials)
+
+- **Personal Access Token (PAT)**  
+  [Learn how to create and use a PAT.](https://zitadel.com/docs/guides/integrate/service-users/authenticate-service-users#personal-access-token)
+
+**Reference:** [Service Users & API Authentication](https://zitadel.com/docs/guides/integrate/service-users/authenticate-service-users#authentication-methods)
+
+---
+
+### 5.2. Import Data with the ZITADEL API
+
+- Use your **access token** or **PAT** to authenticate.
+- Call the [Admin API – Import Data](https://zitadel.com/docs/apis/resources/admin/admin-service-import-data) endpoint, passing your generated JSON file.
+- Verify that the users were imported successfully in the ZITADEL console.
+
+**Import Endpoint:**
+
+- `POST /admin/v1/import`
+- `Authorization: Bearer <token>`
+- **Body:** Generated in step 4.2
```

Comment:
I think we should looking into extending the zitadel tools to automatically upload.

I.e. have a compiled CLI that you can feed with the envs like instance, token and an import file.

Not important now but might be a good improvement.

Reply:
Yes, I absolutely agree. The current tool is just a file formatting tool, not much of a CLI tool. Another point to add as an enhancement request to take a deeper look after the workshop.